python/ray/tune/trial_runner.py

```diff
@@ -385,6 +391,15 @@ def has_resources(self, resources):
         """Returns whether this runner has at least the specified resources."""
         return self.trial_executor.has_resources(resources)
 
+    def _stop_experiment_if_needed(self):
+        """Stops all trials if the user condition is satisfied."""
+        if not self._stopper:
```

Comment:
nit: since the default is `NoopStopper` i think you can get rid of this check right